docs/markdown/Fs-module.md

```diff
@@ -29,3 +31,16 @@ name exists on the file system. This method follows symbolic links.
 
 Takes a single string argument and returns true if the path pointed to
 by the string is a symbolic link.
+
+## Filename modification
+
+### with_suffix
```

Comment:
This is not a great name, because it does not describe what the function actually does. A better name would be something like `set_suffix` or `swap_suffix` depending on what the desired behaviour is. Specifically:

- if there is no existing suffix, what should happen?
- If there are multiple suffixes (`file.a.b.c.`), what should happen?

And so on. The documentation should also explicitly say that this does only string manipulation. No actual files are touched in any way.

Reply:
`with_suffix` just comes from Python [pathlib.Path.with_suffix() method](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix). I have documented and tested these cases now.

as far as name of the fs.with_suffix, the behavior can be:

* adding a suffix if none existed
* changing a suffix
* deleting a suffix

it may be that with_suffix was the least restrictive yet descriptive name of this behavior

Reply:
The C++17 [std::filesystem::path::replace_extension](https://en.cppreference.com/w/cpp/filesystem/path/replace_extension) has the same behavior as .with_suffix().

It does seems like `.replace_suffix()` is more intuitive, at least to me. Or should we stick with .with_suffix()

Reply:
+1 for .replace_suffix(), that's much more obvious as to what it does.

Reply:
OK I changed to .replace_suffix